### PR TITLE
[stable/kafka-pubsub-emulator] Introduce chart for Google Cloud Platform Pub/Sub Emulator for Kafka

### PR DESCRIPTION
**Pull request by riccardomc**
#### What this PR does / why we need it:

This PR introduces an Helm Chart for [Google Cloud Platform Pub/Sub Emulator for Kafka](https://github.com/GoogleCloudPlatform/kafka-pubsub-emulator). See the project GitHub page for additional details.

#### Special notes for your reviewer:

Please notice that in the past I opened a similar PR (https://github.com/helm/charts/pull/9784) that is now stale. This is an updated version of that chart that reflect important changes in the upstream project.

#### Checklist

- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by xiwenc** [comment]
+1

File: stable/kafka-pubsub-emulator/README.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+2. A [Service](https://kubernetes.io/docs/concepts/services-networking/service/) exposing the Pub/Sub Emulator for Kafka
+
+3. A [ConfigMap](https://kubernetes.io/docs/tutorials/configuration/) that holds the Server configuration for the Pub/Sub Emulator for Kafka that will be mounted by the Pods in the StatefulSet above
+
+4. A [PersistentVolume](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) that holds the projects layout (topics and subscriptions) configuration used by the Pub/Sub Emulator for Kafka. 
+
+### Installation
+
+```shell
+helm install --name emulator stable/kafka-pubsub-emulator<Paste>
```
Comment (mattfarina) [inline stable/kafka-pubsub-emulator/README.md]: Did you mean to have `<Paste>` at the end of the line?

File: stable/kafka-pubsub-emulator/templates/statefulset.yaml [hunk 2]
```diff
@@ -0,0 +1,91 @@
+apiVersion: apps/v1beta2
```
Comment (mattfarina) [inline stable/kafka-pubsub-emulator/templates/statefulset.yaml]: Can you please update the `apiVersion` to `apps/v1`. The Kubernetes API will soon stop accepting older versions.

File: stable/kafka-pubsub-emulator/README.md (hunk 1 above)
Comment (riccardomc) [reply stable/kafka-pubsub-emulator/README.md]: I didn't, but apparently my editor thought otherwise... :)

**Comment by riccardomc** [comment]
I see that the test are failing because the emulator application doesn't start. It expects a working kafka cluster, I will add it as a dependency.

**Comment by riccardomc** [comment]
@mattfarina It seems that the chart test times out after around 3 minutes, but I am confident that everything is working properly. The longer deployment time is due to the fact that this charts depends on a kafka cluster to be up and running which in turns expects a zookeeper cluster. This takes time to converge in a ready state. In my tests on minikube it takes around 2:30 minutes. It probably hits the 3 minutes in a real world scenario.

How can I get past this hurdle?

```
% time helm install --name emu1 --namespace emu --wait --timeout 300 .        2019/06/05 11:41:34 Warning: Building values map for chart 'kafka'. Skipped value (map[]) for 'image', as it is not a table.
NAME:   emu1
LAST DEPLOYED: Wed Jun  5 11:41:34 2019
NAMESPACE: emu
STATUS: DEPLOYED

RESOURCES:
==> v1/ConfigMap
NAME                        DATA  AGE
emu1-kafka-pubsub-emulator  1     2m34s

==> v1/Pod(related)
NAME                          READY  STATUS   RESTARTS  AGE
emu1-kafka-0                  1/1    Running  1         2m34s
emu1-kafka-1                  1/1    Running  0         72s
emu1-kafka-2                  1/1    Running  0         38s
emu1-kafka-pubsub-emulator-0  1/1    Running  4         2m34s
emu1-zookeeper-0              1/1    Running  0         2m34s
emu1-zookeeper-1              1/1    Running  0         2m12s
emu1-zookeeper-2              1/1    Running  0         111s

==> v1/Service
NAME                        TYPE       CLUSTER-IP      EXTERNAL-IP  PORT(S)                     AGE
emu1-kafka                  ClusterIP  10.98.204.155   <none>       9092/TCP                    2m34s
emu1-kafka-headless         ClusterIP  None            <none>       9092/TCP                    2m34s
emu1-kafka-pubsub-emulator  ClusterIP  10.110.127.104  <none>       80/TCP                      2m34s
emu1-zookeeper              ClusterIP  10.103.237.203  <none>       2181/TCP                    2m34s
emu1-zookeeper-headless     ClusterIP  None            <none>       2181/TCP,3888/TCP,2888/TCP  2m34s

==> v1/StatefulSet
NAME                        READY  AGE
emu1-kafka-pubsub-emulator  1/1    2m34s

==> v1beta1/PodDisruptionBudget
NAME            MIN AVAILABLE  MAX UNAVAILABLE  ALLOWED DISRUPTIONS  AGE
emu1-zookeeper  N/A            1                1                    2m34s

==> v1beta1/StatefulSet
NAME            READY  AGE
emu1-kafka      3/3    2m34s
emu1-zookeeper  3/3    2m34s


NOTES:
1. Set the PUBSUB_EMULATOR_HOST environment variable:
  export POD_NAME=$(kubectl get pods --namespace emu -l "app.kubernetes.io/name=kafka-pubsub-emulator,app.kubernetes.io/instance=emu1" -o jsonpath="{.items[0].metadata.name}")
  export PUBSUB_EMULATOR_HOST=127.0.0.1:8080
  kubectl port-forward $POD_NAME -- 8080:8080

2. Google Pub/Sub Client libraries will automatically use the address in
the PUBSUB_EMULATOR_HOST environment variable.

helm install --name emu1 --namespace emu --wait --timeout 300 .  0.10s user 0.02s system 0% cpu 2:34.51 total
```

**Comment by maorfr** [comment]
please take a look at helm/helm#6095 (comment)

we strongly encourage people to self host chart repos:
https://github.com/helm/hub/blob/master/Repositories.md

you can even do that using GitHub pages:
https://github.com/helm/hub/blob/master/docs/github.md#github-pages-as-a-repository

**Comment by riccardomc** [comment]
@maorfr thanks for the info. Moving this chart to https://github.com/ContainerSolutions/helm-charts/tree/master/charts/kafka-pubsub-emulator then

**Pull request closed without merging**